src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs

```diff
@@ -463,15 +463,40 @@ private static void SetGlobalOptions(IList<string> args, ChocolateyConfiguration
         private static void SetProxyOptions(ChocolateyConfiguration config, Container container)
         {
             // Evaluation order of Proxy settings: System Set -> Environment Variable Set -> Chocolatey Configuration File Set -> CLI Passed in Argument
-            // If we don't yet have a Proxy Location, check if the system has one configured in the registry
-            if (string.IsNullOrWhiteSpace(config.Proxy.Location) && Platform.GetPlatform() == PlatformType.Windows)
+            var proxyAlreadySet = !string.IsNullOrWhiteSpace(config.Proxy.Location);
+            var onWindows = Platform.GetPlatform() == PlatformType.Windows;
+
+            // Only Windows has a registry provider, if it's already set, or we're not on Windows we don't need to continue.
+            if (proxyAlreadySet || !onWindows)
             {
-                var registryService = container.GetInstance<IRegistryService>();
-                var internetSettingsRegKey = registryService.GetKey(RegistryHive.CurrentUser, "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Internet Settings");
-                if (internetSettingsRegKey.GetValue("ProxyEnable").ToStringSafe().IsEqualTo("1"))
+                return;
+            }
+
+            // We don't yet have a Proxy Location, check if the system has one configured in the registry
+            var registryService = container.GetInstance<IRegistryService>();
+            var internetSettingsRegKey = registryService.GetKey(RegistryHive.CurrentUser, "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Internet Settings");
+
+            if (internetSettingsRegKey.GetValue("ProxyEnable").ToStringSafe().IsEqualTo("1"))
+            {
+                var proxySetting = internetSettingsRegKey.GetValue("ProxyServer").ToStringSafe();
+
+                if (string.IsNullOrWhiteSpace(proxySetting))
+                {
+                    return;
+                }
+
+                if (proxySetting.IndexOf(';') != -1)
                 {
-                    config.Proxy.Location = internetSettingsRegKey.GetValue("ProxyServer").ToStringSafe();
+                    var allProxies = proxySetting.Split(';');
+                    proxySetting = allProxies.FirstOrDefault(s => s.TrimSafe().StartsWith("https="))?.Replace("https=", string.Empty);

```

Comment:
Using .Replace here feels _weird_ to me just knowing it _could_ do a substitution somewhere other than the start of the string... I guess it should be fine, I'm curious if it would be a ton more complicated to do like a weird `IndexOf` and then `Substring` to get the remainder instead though... not sure it's worth doing that though.

Reply:
Ooh, wait. Maybe you could do `IndexOf("https=") == 0` instead of the StartsWith, and then do a `.Split('=')[1]`

I think that'd work, and _might_(?) be cheaper than a .Replace()?